core/src/qos_service.rs

```diff
@@ -119,13 +121,14 @@ impl QosService {
     // invoke cost_model to calculate cost for the given list of transactions
     fn compute_transaction_costs<'a>(
         &self,
+        feature_set: &FeatureSet,
         transactions: impl Iterator<Item = &'a SanitizedTransaction>,
     ) -> Vec<TransactionCost> {
         let mut compute_cost_time = Measure::start("compute_cost_time");
         let cost_model = self.cost_model.read().unwrap();
```

Comment:
no need to lock anymore or even have an instance. Can use `CostModel::calculate_cost(...)` below